internal/venice-common/src/main/java/com/linkedin/venice/pushmonitor/PushStatusDecider.java

```diff
@@ -272,34 +282,29 @@ protected ExecutionStatus getPartitionStatus(
         if (!currentStatus.equals(COMPLETED)) {
           isLeaderCompleted = false;
         }
-
-        if (currentStatus.equals(ERROR)) {
-          isLeaderInError = true;
+        // The following has a potential to mark all or multiple replicas as disabled which could lead
```

Comment:
I thought about this again and with the new change, there will be a behavior change.
In the past, if ingestion failed  in multiple nodes because of other reasons, such as disk full or Kafka/ZK issue, the push job will fail, but now, all the jobs will become stuck, which may cause a lot of confusions..
So the goal of this PR is to tolerate single point failure, and if there are multiple node failures, the push job should fail as usual.
So maybe checking whether the offline replica is disabled or not is necessary, and if we have concerns around the requests to ZK, we could do follows:
For a given checkPushStatusAndDetailsByPartitionsStatus invocation, we only look up instanceConfig once per problematic instance (instance has offline replica).
Throttle the qps by an event throttler for getDisabledPartitionsMap calls.
I think they will achieve both goals:
Handle SPoF.
Maintain the same operational strategy.
Not overwhelming ZK (normally, the throttler should kick in since offline replicas shouldn't be common).
Metric/Alert needs to be setup anyway.